src/lints/manual/manual_unwrap_or_else.rs

```diff
@@ -0,0 +1,301 @@
+
+#[tracing::instrument(skip_all, level = "trace")]
+pub fn check_manual_unwrap_or_else<'db>(
+    db: &'db dyn Database,
+    item: &ModuleItemId<'db>,
+    diagnostics: &mut Vec<PluginDiagnostic<'db>>,
+) {
+    let function_bodies = get_all_function_bodies_with_ids(db, item);
+
+    for (function_id, function_body) in function_bodies {
+        let if_exprs = get_all_if_expressions(function_body);
+        let match_exprs = get_all_match_expressions(function_body);
+        let arenas = &function_body.arenas;
+
+        for match_expr in match_exprs {
+            if check_manual_unwrap_or_else_with_match(db, &match_expr, function_id, arenas) {
+                diagnostics.push(PluginDiagnostic {
+                    stable_ptr: match_expr.stable_ptr.untyped(),
+                    message: ManualUnwrapOrElse.diagnostic_message().to_owned(),
+                    severity: Severity::Warning,
+                    inner_span: None,
+                });
+            }
+        }
+
+        for if_expr in if_exprs {
+            if check_manual_unwrap_or_else_with_if(db, &if_expr, function_id, arenas) {
+                diagnostics.push(PluginDiagnostic {
+                    stable_ptr: if_expr.stable_ptr.untyped(),
+                    message: ManualUnwrapOrElse.diagnostic_message().to_owned(),
+                    severity: Severity::Warning,
+                    inner_span: None,
+                });
+            }
+        }
+    }
+}
+
+fn check_manual_unwrap_or_else_with_match<'db>(
+    db: &'db dyn Database,
+    match_expr: &ExprMatch<'db>,
+    function_id: FunctionWithBodyId<'db>,
+    arenas: &Arenas<'db>,
+) -> bool {
+    let matched_expr = db.expr_semantic(function_id, match_expr.matched_expr);
+    let is_droppable = db.droppable(matched_expr.ty()).is_ok();
+    let is_manual_unwrap_or_else =
+        check_manual(db, match_expr, arenas, ManualLint::ManualUnwrapOrElse);
+    is_manual_unwrap_or_else && !is_droppable
+}
+
+fn check_manual_unwrap_or_else_with_if<'db>(
+    db: &'db dyn Database,
+    if_expr: &ExprIf,
+    function_id: FunctionWithBodyId<'db>,
+    arenas: &Arenas<'db>,
+) -> bool {
+    let condition_expr = db.expr_semantic(function_id, if_expr.if_block);
+    let is_droppable = db.droppable(condition_expr.ty()).is_ok();
+    let is_manual_unwrap_or_else =
+        check_manual_if(db, if_expr, arenas, ManualLint::ManualUnwrapOrElse);
+    is_manual_unwrap_or_else && !is_droppable
+}
+
+#[tracing::instrument(skip_all, level = "trace")]
+fn fix_manual_unwrap_or_else<'db>(
+    db: &'db dyn Database,
+    node: SyntaxNode<'db>,
+) -> Option<InternalFix<'db>> {
+    let expr = ast::Expr::from_syntax_node(db, node);
+
+    let (matched_expr, or_body) = match &expr {
+        ast::Expr::Match(expr_match) => {
+            let mut arms = expr_match.arms(db).elements(db);
+            let matched_expr = expr_match.expr(db).as_syntax_node();
+            let arm = arms.nth(1).expect("Expected a `match` with second arm.");
```

Comment:
```suggestion
            let arm = arms.nth(1)?;
```
We didn't check this earlier (correct me if I'm wrong), so there is no guarantee it won't panic

Reply:
Actually we should **expect** the match expr to have a second arm, as it's checked [here](https://github.com/software-mansion/cairo-lint/blob/79571606e24efb61c963ea3f5262739683d10380/src/lints/manual/mod.rs#L66). The overall concept of fixers, is that we can assume a lot of things, as the syntax node in the args, is coming based on the diagnostics that we sent from the code above. 

tl;dr
We should expect certain type/form of data in the fixer. If it panics, it means that the diagnostics above are doing something wrong